src/core/autosave/autosave.hook.tsx

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useRef } from 'react';
+
+import {
+  useCanvasSchemaContext,
+  useCanvasViewSettingsContext,
+} from '@/core/providers';
+
+import { saveToLocal, retrieveFromLocal } from './autosave.business';
+import { retrieveValueFromLocalStorage } from '@/common/local-storage';
+
+const useAutosave = () => {
+  const { canvasSchema, setCanvasSchema } = useCanvasSchemaContext();
+  const { filename, setFilename, setLoadSample } =
+    useCanvasViewSettingsContext();
+
+  const INTERVAL = 6000;
+  const AUTOSAVE_KEY = 'autoSaveFile';
+
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    const retrieveSchema = retrieveFromLocal(AUTOSAVE_KEY, setLoadSample);
+    const retrieveFileName = retrieveValueFromLocalStorage(AUTOSAVE_KEY);
+
+    if (retrieveFileName && retrieveFileName.filename !== '')
+      setFilename(retrieveFileName.filename);
+
+    setCanvasSchema(retrieveSchema);
+  }, []);
+
+  useEffect(() => {
+    const autosaveHandler = () => {
+      if (canvasSchema.tables.length !== 0) {
+        saveToLocal(AUTOSAVE_KEY, {
+          filename: filename ?? undefined,
+          canvasSchema,
+        });
+      }
+      if (canvasSchema.tables.length === 0) {
+        localStorage.removeItem(AUTOSAVE_KEY);
+        setLoadSample(true);
+      }
+    };
+    timerRef.current = setInterval(autosaveHandler, INTERVAL);
+
+    return () => {
+      if (timerRef.current) {
+        clearInterval(timerRef.current);
+        timerRef.current = null;
+      }
+    };
+  }, [canvasSchema, filename]);
+
+  const stopAutosave = () => {
+    if (timerRef.current) {
+      clearInterval(timerRef.current);
+      timerRef.current = null;
```

Comment:
Missing trying to save on close

Reply:
Is this possible?  what I think is happening is that `stopAutosave` will be executed when component has already been unmounted, so it doesn't have access to `canvasSchema` to save it

Reply:
That's my doubt, well the provider is created at App level, I think it would be a good idea to give a try and check what happens

Reply:
Moved to App level, and still unable to get it to work